source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.cpp

```diff
@@ -0,0 +1,638 @@
+	setDetectorTotalGain(newValue);
+}
+
+void AmptekSDD123EPICSDetectorManager::onPileUpRejectionControlValueChanged(double newValue){
+	if(!connected_ || !receivedConfigurationDataOnce_)
+		return;
+	if(newValue == 0 && configurationData_.pileUpRejection_ == "OFF")
+		return;
+	else if(newValue == 1 && configurationData_.pileUpRejection_ == "ON")
+		return;
+	else if(newValue == 0)
+		setDetectorPileUpRejection(false);
+	else
+		setDetectorPileUpRejection(true);
+}
+
+void AmptekSDD123EPICSDetectorManager::onThermoelectricCoolerControlValueChanged(double newValue){
+	if(!connected_ || !receivedConfigurationDataOnce_)
+		return;
+	if(configurationData_.coolerSetting_ == "OFF"){
+		//I need to do something about this
+		return;
+	}
+	double coolerSetting = configurationData_.coolerSetting_.toDouble();
+	if(thermoelectricCoolerControl_->withinTolerance(coolerSetting))
+		return;
+	setDetectorCoolerSetting((int)newValue);
+}
+
+void AmptekSDD123EPICSDetectorManager::onFastThresholdControlValueChanged(double newValue){
+	if(!connected_ || !receivedConfigurationDataOnce_)
+		return;
+	if(fastThresholdControl_->withinTolerance(configurationData_.fastThreshold_))
+		return;
+	setDetectorFastThreshold((int)newValue);
+}
+
+void AmptekSDD123EPICSDetectorManager::onSlowThresholdControlValueChanged(double newValue){
+	if(!connected_ || !receivedConfigurationDataOnce_)
+		return;
+	if(slowThresholdControl_->withinTolerance(configurationData_.slowThreshold_))
+		return;
+	setDetectorSlowThreshold(newValue);
+}
+
+void AmptekSDD123EPICSDetectorManager::onPeakingTimeControlValueChanged(double newValue){
+	if(!connected_ || !receivedConfigurationDataOnce_)
+		return;
+	if(peakingTimeControl_->withinTolerance(configurationData_.peakingTime_))
+		return;
+	setDetectorPeakingTime(newValue);
+}
+
+void AmptekSDD123EPICSDetectorManager::onFastChannelPeakingTimeControlValueChanged(double newValue){
+	if(!connected_ || !receivedConfigurationDataOnce_)
+		return;
+	if(newValue == 0 && configurationData_.fastChannelPeakingTime_ == 50)
+		return;
+	else if(newValue == 1 && configurationData_.fastChannelPeakingTime_ == 100)
+		return;
+	else if(newValue == 2 && configurationData_.fastChannelPeakingTime_ == 400)
+		return;
+	else{
+		int enumValue = (int)newValue;
+		switch(enumValue){
+		case 0:
+			setDetectorFastPeakingTime(AmptekSDD123DetectorManager::FastPeakingTime50);
+			break;
+		case 1:
+			setDetectorFastPeakingTime(AmptekSDD123DetectorManager::FastPeakingTime100);
+			break;
+		case 2:
+			setDetectorFastPeakingTime(AmptekSDD123DetectorManager::FastPeakingTime400);
+			break;
+		default:
+			break;
+		}
+	}
+}
+
+void AmptekSDD123EPICSDetectorManager::onROI1LowIndexControlValueChanged(double newValue){
+	if(!connected_ || !receivedConfigurationDataOnce_)
+		return;
+	if(roi1LowIndexControl_->withinTolerance(configurationData_.scaLowIndices_.at(0)))
+		return;
+	setDetectorSingleChannelAnalyzer(1, (int)newValue, configurationData_.scaHighIndices_.at(0));
```

Comment:
@dretrex yes, I think you are right, although the old codes are like this as well. @davidChevrier I believe that this is a bug. I will fix it in AMDS